File: app/src/main/java/linusfessler/alarmtiles/alarm/AlarmDao.java
package linusfessler.alarmtiles.alarm;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface AlarmDao {

    @Query("SELECT * FROM Alarm")
    LiveData<Alarm> select();

    @Insert
    void insert(Alarm alarm);

    @Update
    void update(Alarm alarm);

    @Query("SELECT COUNT(*) FROM Alarm")
    int count();
}

File: app/src/main/java/linusfessler/alarmtiles/alarm/AlarmDao.kt
package linusfessler.alarmtiles.alarm

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import io.reactivex.Observable

@Dao
interface AlarmDao {
    @Query("SELECT * FROM Alarm")
    fun select(): Observable<Alarm>

    @Insert
    fun insert(alarm: Alarm)

    @Update
    fun update(alarm: Alarm)

    @Query("SELECT COUNT(*) FROM Alarm")
    fun count(): Int
}